### PR TITLE
kentutils: update to @459, update download location & deps

### PR DESCRIPTION
**Pull request by A-N-Other**
- Updating to `@459`.
- The download location was incorrectly pointing to https://github.com/ENCODE-DCC/kentUtils, which is an archive of a version from 2014! I've corrected it to the proper location: https://hgdownload.cse.ucsc.edu/admin/exe/userApps.archive/.
- Added missing dependencies.

Have made a note in the `package.py` for the next person, but I've deprecated the existing version as that location (and hence version) should never have been the one chosen for this package. Suggest the old version and its conflicts are dropped on the next update...

### CODE REVIEW

File: var/spack/repos/builtin/packages/kentutils/package.py [hunk 1]
```diff
@@ -9,26 +9,40 @@
...
+    depends_on("libiconv")
+
+    conflicts("%cce", when="@302.1")
+    conflicts("%apple-clang", when="@302.1")
+    conflicts("%clang", when="@302.1")
+    conflicts("%intel", when="@302.1")
+    conflicts("%nag", when="@302.1")
+    conflicts("%pgi", when="@302.1")
+    conflicts("%xl", when="@302.1")
+    conflicts("%xl_r", when="@302.1")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/kentutils/package.py]:
```suggestion
    requires("%gcc", when="@302.1")
```

File: var/spack/repos/builtin/packages/kentutils/package.py [hunk 2]
```diff
@@ -9,26 +9,40 @@
...
+
+    def setup_build_environment(self, env):
+        # Builds fall over without this being manually specified
+        env.append_flags("LDFLAGS", f'{self.spec["libiconv"].libs.ld_flags}')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/kentutils/package.py]: Can we use `flag_handler`?

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/kentutils/package.py (hunk 1 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/kentutils/package.py]: Far neater!

File: var/spack/repos/builtin/packages/kentutils/package.py (hunk 2 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/kentutils/package.py]: We can! See separate commit.

**Review by alalazo — APPROVED** [review]

**Pull request merged**